src/CommunityToolkit.Maui/PopupService.cs

```diff
@@ -0,0 +1,171 @@
+﻿using CommunityToolkit.Maui.Core;
+using CommunityToolkit.Maui.Views;
+using Microsoft.Maui.Controls.Platform;
+using System.ComponentModel;
+
+namespace CommunityToolkit.Maui;
+
+/// <inheritdoc cref="IPopupService"/>
+public class PopupService : IPopupService
+{
+	readonly IServiceProvider serviceProvider;
+
+	static readonly IDictionary<Type, Type> viewModelToViewMappings = new Dictionary<Type, Type>();
```

Comment:
since this is private and not injected, we should use the concrete type, changing in the future, if needed, will not be a breaking change.
```suggestion
	static readonly Dictionary<Type, Type> viewModelToViewMappings = new Dictionary<Type, Type>();
```